datafusion-storage/src/lib.rs

```diff
@@ -18,7 +18,10 @@
 pub mod object_store;
 
 use chrono::{DateTime, Utc};
-use datafusion_common::ScalarValue;
+use std::{io, result};
+
+/// Result type for operations that could result in an io error
+pub type Result<T> = result::Result<T, io::Error>;
```

Comment:
Im wondering if any impact from using this instead of `DataFusionError::IoError` but i think its fine and of course it would require keeping datafusion-common dependency which would defeat the point of this PR.

Reply:
The goal of this PR is to remove the datafusion-common dependency. 

When you need to use DataFusionError::IoError, it's simply to add *.map_err(DataFusionError::IoError)* like some places in this PR does.